crates/pallet-domains/src/lib.rs

```diff
@@ -1493,31 +1490,14 @@ impl<T: Config> Pallet<T> {
             FraudProofError::ChallengingGenesisReceipt
         );
 
-        if let FraudProof::InvalidBundle(InvalidBundleProof::TotalRewards(
-            InvalidTotalRewardsProof {
-                domain_id,
-                block_number,
-                domain_block_hash,
-                storage_proof,
-                ..
-            },
-        )) = fraud_proof
+        if let FraudProof::InvalidTotalRewards(InvalidTotalRewardsProof { storage_proof, .. }) =
```

Comment:
Perhaps we should move the verification logic to `sp_domains` like:
```rust
impl InvalidTotalRewardsProof {
    pub fn verify(&self, bad_receipt: ExecutionReceipt) -> bool {
        if bad_receipt.hash() != self.bad_receipt_hash {
            return false
        }
        ...
    }
}
```
to keep `pallet-domains` from bloating, non-blocker.

Reply:
My point is to keep `pallet-domains` from bloating, moving to `sp_domains` the verification can still be used in runtime.